gazebo_plugins/src/gazebo_ros_planar_move.cpp

```diff
@@ -232,17 +301,37 @@ namespace gazebo
     odom_.pose.pose.orientation.y = pose.rot.y;
     odom_.pose.pose.orientation.z = pose.rot.z;
     odom_.pose.pose.orientation.w = pose.rot.w;
-    odom_.pose.covariance[0] = 0.00001;
-    odom_.pose.covariance[7] = 0.00001;
     odom_.pose.covariance[14] = 1000000000000.0;
     odom_.pose.covariance[21] = 1000000000000.0;
     odom_.pose.covariance[28] = 1000000000000.0;
-    odom_.pose.covariance[35] = 0.001;
 
     // get velocity in /odom frame
     math::Vector3 linear;
     linear.x = (pose.pos.x - last_odom_pose_.pos.x) / step_time;
     linear.y = (pose.pos.y - last_odom_pose_.pos.y) / step_time;
+
+    if (static_x_variance_)
+    {
+      odom_.pose.covariance[0] = x_variance_;
+    }
+    else
+    {
+      odom_.pose.covariance[0] += calculateScaledVelocityVariance(linear.x,
```

Comment:
No, but that's realistic. The robot's raw odometry covariance *should* be unbounded.